consensus-starfish/core/src/encoder.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use reed_solomon_simd::ReedSolomonEncoder;
+
+use crate::{Transaction, block::Shard};
+
+pub type Encoder = ReedSolomonEncoder;
+
+pub trait ShardEncoder {
+    fn encode_shards(
+        &mut self,
+        data: Vec<Shard>,
+        info_length: usize,
+        parity_length: usize,
+    ) -> Vec<Shard>;
+
+    #[allow(dead_code)]
+    fn encode_statements(
+        &mut self,
+        block: Vec<Transaction>,
+        info_length: usize,
+        parity_length: usize,
+    ) -> Vec<Shard>;
+}
+
+impl ShardEncoder for Encoder {
+    fn encode_shards(
+        &mut self,
+        mut data: Vec<Shard>,
+        info_length: usize,
+        parity_length: usize,
+    ) -> Vec<Shard> {
+        assert_eq!(
+            data.len(),
+            info_length,
+            "Data length must match info length"
+        );
+        assert!(info_length > 0, "Info length must be greater than 0");
+        let shard_bytes = data[0].len();
+        self.reset(info_length, parity_length, shard_bytes)
+            .expect("Reset failed");
```

Comment:
stay in line with other messages 
```suggestion
            .expect("reset should not fail");
```